### DISCUSSION TITLE
About Identifying the Other class in Model for Unknown Images...

### DISCUSSION THREAD

**Discussion by VISHAL1855** [mrdbourke/pytorch-deep-learning#156] [General]
Hii all As we have created a Model it assumes or predicts three classes. If I gave a Input image as any other unrelated image I should get as a Other class or not related Image. So prior to our problem we can do a softmax threshholding i.e select a threshold as 0.75 Or 0.80 if prediction is greater than threshold than It will classify as Our class and If prediction is less than threshold it should classify as Other class .
So Can Anyone help me in Implementing the Idea in code. I am Not able to code for the problem can anyone help me out.. @mrdbourke @beneyal @schafsam @aronvandepol

**Comment by aronvandepol** [comment]
While @mrdbourke certainly has more insight into this problem than me. I can conceive of two solutions.

- Add a 4th class (seems like a bad idea, as you'd have to train it too).

- Manually check if the probabilities for all classes is below your set treshold. If it is, return as class 'other'. You can check the probability before using softmax, but after using argmax.

**Reply by VISHAL1855** [reply]
But in Gradio app we need to return a dictionaty how can I return there Other class ?

**Reply by aronvandepol** [reply]
You can manually add this class in your dictionary.

**Comment by mrdbourke** [comment]
Hey @VISHAL1855,

You are correct in your thinking of thresholding.

You could make it so if the model classifies the image as <0.75 or <0.80 or some other number, it returns an "unknown" value.

You could also do what @aronvandepol suggested and add another class such as "not food" or "unrelated" for when an image isn't of food or even a fourth class for some other object.
